pkg/session/session.go

```diff
@@ -30,88 +31,155 @@ type session struct {
-
 	if err != nil {
 		return nil, fmt.Errorf("error connecting to cluster as admin: %w", err)
 	}
-
 	s.client = client
 	s.adminClient = adminClient
 	s.isAuthenticated = true
-	currentSession = s
 	return client, nil
 }
 
-func (s session) Disconnect() {
+func (s *session) Disconnect() {
 	if s.client != nil {
 		s.client.Close()
 	}
 	s.client = nil
 	s.adminClient = nil
 	s.isAuthenticated = false
-	currentSession = s
 	fmt.Println("Logged out successfully!")
 }
 
-func (s session) IsAuthenticated() bool {
+func (s *session) IsAuthenticated() bool {
 	return s.isAuthenticated
 }
 
-func (s session) GetClient() sarama.Client {
-	if s.client == nil {
-		return nil
+func (s *session) GetClient() (sarama.Client, error) {
+	if s.client != nil {
+		return s.client, nil
 	}
-	return s.client
-}
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
 
-func (s session) GetAdminClient() sarama.ClusterAdmin {
-	if s.adminClient == nil {
-		return nil
+	client, err := s.Connect(ctx)
+	if err != nil {
+		return nil, err
 	}
+	return client, nil
+}
 
-	return s.adminClient
+func (s *session) GetAdminClient() (sarama.ClusterAdmin, error) {
+	if s.adminClient != nil {
+		return s.adminClient, nil
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	adminClient, err := cluster.NewCluster(s.brokers).ConnectAdmin(ctx)
+	if err != nil {
+		return nil, err
+	}
+	s.adminClient = adminClient
+	return adminClient, nil
 }
 
-func GetCurrentSession() Session {
+func GetCurrentSession() *session {
 	return currentSession
 }
 
-var currentSession Session
+var currentSession *session
+
+func createDefaultSession() error {
+	file, err := os.Create(constants.OpenKommanderConfigFilename)
+	if err != nil {
+		fmt.Println("Error creating session file:", err)
+		return err
+	}
+	defer file.Close()
+
+	sessionData := SessionData{Brokers: []string{}, IsAuthenticated: false}
+	return json.NewEncoder(file).Encode(sessionData)
+}
+
+func saveSession() error {
+	file, err := os.Create(constants.OpenKommanderConfigFilename)
+	if err != nil {
+		fmt.Println("Error creating session file:", err)
+		return err
+	}
+	defer file.Close()
+
+	sessionData := SessionData{Brokers: currentSession.brokers, IsAuthenticated: currentSession.isAuthenticated}
+	err = json.NewEncoder(file).Encode(sessionData)
+	if err != nil {
+		fmt.Println("Error encoding session data:", err)
+		return err
+	}
+	return nil
+}
+
+func loadSession() error {
+	file, err := os.Open(constants.OpenKommanderConfigFilename)
+	if err != nil {
+		err = createDefaultSession()
+		if err != nil {
+			fmt.Println("Error creating session file:", err)
+			return err
+		}
+
+		file, err = os.Open(constants.OpenKommanderConfigFilename)
+		if err != nil {
+			fmt.Println("Error opening session file:", err)
+			return err
+		}
+	}
+	defer file.Close()
+
+	var data SessionData
+	decoder := json.NewDecoder(file)
+	err = decoder.Decode(&data)
```

Comment:
can't you decode directly to `&currentSession` without the intermediate `data` object?